tsdb/index/tsi1/partition.go

```diff
@@ -452,7 +452,9 @@ func (p *Partition) RetainFileSet() (*FileSet, error) {
 }
 
 func (p *Partition) retainFileSet() *FileSet {
```

Comment:
Do you think it might be better to make a `Clone()` method associated with `(*FileSet)` that does the copy?  That way the details of copying a `FileSet` remain near the implementation of `FileSet`.